### PR TITLE
Dscs 457

### PR DESCRIPTION
**Pull request by snezanam**
algorithm field for copy and  support for carriers having a default algorithm, @shuangzhou , @terenceingram could you please have a look.

### CODE REVIEW

File: src/amberdb/sql/Lookups.java [hunk 1]
```diff
@@ -226,6 +226,31 @@ protected abstract void seedLookupTable(@Bind("id") List<Long> id,
                                               @Bind("value") List<String> value,
                                               @Bind("deleted") List<String> deleted);
     
+    
+    @SqlQuery("select distinct linkId, name, carrierId, algorithmId,  from carrier_algorithm where name = :name")
+    public abstract List<CarrierAlgorithm> findCarrierAlgorithmsByName(@Bind("name") String name);
```
Comment (shuangzhou) [inline src/amberdb/sql/Lookups.java]: @snezanam might need to register a CarrierAlgorithmMapper here.

File: src/amberdb/sql/Lookups.java [hunk 2]
```diff
@@ -226,6 +226,31 @@ protected abstract void seedLookupTable(@Bind("id") List<Long> id,
...
+    
+    @SqlQuery("select distinct linkId, name, carrierId, algorithmId  from carrier_algorithm where name = :name and carrierId = :carrierId")
+    public abstract List<CarrierAlgorithm> findCarrierAlgorithmByNameAndId(@Bind("name") String name, @Bind("carrierId")Long carrierId );
```
Comment (shuangzhou) [inline src/amberdb/sql/Lookups.java]: @snezanam mapper might also be required here.

File: src/amberdb/sql/LookupsSchema.java [hunk 3]
```diff
@@ -23,6 +23,17 @@
             + "deleted   varchar(1) default 'N' )")
     public abstract void createLookupTable();
     
+    /*
+     * Link Lookups table - stores relationships between 2 lookups
+     */
+    @SqlUpdate(
+            "CREATE TABLE IF NOT EXISTS carrier_algorithm ("
+            + "linkId        bigint(11), "
```
Comment (shuangzhou) [inline src/amberdb/sql/LookupsSchema.java]: @snezanam might be good to add an index on linkId.

File: src/amberdb/sql/LookupsSchema.java [hunk 4]
```diff
@@ -23,6 +23,17 @@
...
+    /*
+     * Link Lookups table - stores relationships between 2 lookups
+     */
+    @SqlUpdate(
+            "CREATE TABLE IF NOT EXISTS carrier_algorithm ("
+            + "linkId        bigint(11), "
+            + "name      varchar(50), "
+            + "carrierId       bigint(11), "
+            + "algorithmId       bigint(11) )")
+    public abstract void createCarrierAlgorithmTable();
```
Comment (shuangzhou) [inline src/amberdb/sql/LookupsSchema.java]: @snezanam might be good to add a combined index on (carrierId, algorithmId)

**Comment by shuangzhou** [comment]
Looks good apart from above comments.

**Comment by snezanam** [comment]
Thanks Shuang, I have made suggested changes, please have a look.

**Comment by shuangzhou** [comment]
Looks fine to me, might be good to have a unit test to illustrate how CarrierAlgorithm will be accessed and validated.  Good work.

**Pull request merged**